src/Symfony/Component/Validator/Constraints/BicValidator.php

```diff
@@ -93,5 +104,39 @@ public function validate($value, Constraint $constraint)
 
             return;
         }
+
+        // check against an IBAN
+        $iban = $constraint->iban;
+        $path = $constraint->ibanPropertyPath;
+        if ($path && null !== $object = $this->context->getObject()) {
+            try {
+                $iban = $this->getPropertyAccessor()->getValue($object, $path);
+            } catch (NoSuchPropertyException $e) {
+                throw new ConstraintDefinitionException(sprintf('Invalid property path "%s" provided to "%s" constraint: %s', $path, \get_class($constraint), $e->getMessage()), 0, $e);
+            }
+        }
+        if (!$iban) {
+            return;
+        }
+        $ibanCountryCode = substr($iban, 0, 2);
+        if (ctype_alpha($ibanCountryCode) && substr($canonicalize, 4, 2) !== $ibanCountryCode) {
+            $this->context->buildViolation($constraint->ibanMessage)
+                ->setParameter('{{ value }}', $this->formatValue($value))
+                ->setParameter('{{ iban }}', $iban)
+                ->setCode(Bic::INVALID_IBAN_COUNTRY_CODE_ERROR)
+                ->addViolation();
+        }
+    }
+
+    private function getPropertyAccessor(): PropertyAccessor
+    {
+        if (null === $this->propertyAccessor) {
+            if (!class_exists('Symfony\Component\PropertyAccess\PropertyAccess')) {
```

Comment:
If I don't miss anything, this cannot happen as the constraint would have thrown the exception before.

Reply:
that would apply to all constraints where we move the check: #28644

Reply:
@nicolas-grekas should I then remove the exception here as it's already thrown in the constraint constructor?

Reply:
@nicolas-grekas I think this `getPropertyAccessor()` should be remove: we either use DI in the constructor to get the `PropertyAccess` component or use the static way to get it (like for Intl)

Reply:
the `BicValidator` has no `instanceof Bic` test (should be added though for consistency).

Technically this throw can still happen if called before `validate()` is called (unlikely, but possible from API pov)

I think it makes sense to keep the check; it's more easy to reason about, causes less confusion and makes no assumptions. And all constraints do it already, we can keep it even in 5.0. That's the shortest path towards consistency concerning accessing deps.